app/controllers/users/passwords_controller.rb

```diff
@@ -6,12 +6,27 @@ class Users::PasswordsController < Devise::PasswordsController
   layout "session"
 
   def create
-    if User.find_for_authentication(resource_params.permit(:login))&.protected_user
+    user = User.find_for_authentication(resource_params.permit(:login))
+
+    if user&.protected_user
       flash[:error] = t(".reset_blocked", contact_abuse_link: view_context.link_to(t(".contact_abuse"), new_abuse_report_path)).html_safe
       redirect_to root_path and return
+    elsif user&.password_resets_limit_reached?
+      available_time = ApplicationController.helpers.time_in_zone(
+        user.password_resets_available_time, nil, user
+      )
+
+      flash[:error] = t(".reset_cooldown", reset_available_time: available_time).html_safe
+      redirect_to root_path and return
     end
+
     super do |user|
-      flash.now[:notice] = ts("We couldn't find an account with that email address or username. Please try again?") if user.nil? || user.new_record?
+      if user.nil? || user.new_record?
+        flash.now[:notice] = ts("We couldn't find an account with that email address or username. Please try again?")
+      else
+        user.update_password_resets_requested
```

Comment:
Would it make sense to shift this into the `User` model, something like the following?
```ruby
after_update :update_password_resets_requests, if: :will_save_change_to_password?
```

Reply:
I don't believe so, since `resets_requested` is updated when the reset email is sent rather than when the password is changed.

Reply:
Ah, yes, I think I got those 2 mixed together in my head. Feel free to mark this thread as resolved!